src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -383,6 +383,46 @@ public StringSegment Trim()
         }
 
         /// <summary>
+        /// Removes all leading whitespaces.
+        /// </summary>
+        /// <returns>The trimmed <see cref="StringSegment"/>.</returns>
+        public StringSegment TrimStart()
```

Comment:
Hmm I thought although structs have smaller allocation costs, it is still non-zero.